starport/cmd/network_chain_show.go

```diff
@@ -0,0 +1,181 @@
+)
+
+var (
+	showTypes = map[ShowType]struct{}{
+		chainShowInfo:     {},
+		chainShowGenesis:  {},
+		chainShowAccounts: {},
+		chainShowPeers:    {},
+	}
+
+	chainAccSummaryHeader = []string{"Genesis Account", "Coins"}
+)
+
+// NewNetworkChainShow creates a new chain show command to show
+// a chain on SPN.
+func NewNetworkChainShow() *cobra.Command {
+	c := &cobra.Command{
+		Use:   "show [info|genesis|accounts|peers] [launch-id]",
+		Short: "Show details of a chain",
+		RunE:  networkChainShowHandler,
+		Args:  cobra.ExactArgs(2),
+	}
+
+	c.Flags().AddFlagSet(flagSetKeyringBackend())
+	c.Flags().AddFlagSet(flagNetworkFrom())
+	c.Flags().AddFlagSet(flagSetHome())
+
+	return c
+}
+
+func networkChainShowHandler(cmd *cobra.Command, args []string) error {
+	showType := ShowType(args[0])
+	if _, ok := showTypes[showType]; !ok {
+		cmd.Usage()
+		return fmt.Errorf("invalid arg %s", showType)
+	}
+
+	nb, err := newNetworkBuilder(cmd)
+	if err != nil {
+		return err
+	}
+	defer nb.Cleanup()
+
+	// parse launch ID.
+	launchID, err := network.ParseLaunchID(args[1])
+	if err != nil {
+		return err
+	}
+
+	n, err := nb.Network()
+	if err != nil {
+		return err
+	}
+
+	chainLaunch, err := n.ChainLaunch(cmd.Context(), launchID)
+	if err != nil {
+		return err
+	}
+
+	c, err := nb.Chain(networkchain.SourceLaunch(chainLaunch))
+	if err != nil {
+		return err
+	}
+
+	content := ""
+	switch showType {
+	case chainShowGenesis:
+		content, err = formatChainGenesis(c)
+	case chainShowInfo:
+		content, err = formatChainInfo(cmd.Context(), c, launchID)
+	case chainShowAccounts:
+		content, err = formatChainAccounts(cmd.Context(), n, launchID)
+	case chainShowPeers:
+		content, err = formatChainPeers(cmd.Context(), n, launchID)
+	}
+	if err != nil {
+		return err
+	}
+
+	nb.Spinner.Stop()
+	fmt.Print(content)
+	return nil
+}
+
+func formatChainGenesis(c network.Chain) (string, error) {
+	genesisPath, err := c.GenesisPath()
+	if err != nil {
+		return "", err
+	}
+	if _, err = os.Stat(genesisPath); os.IsNotExist(err) {
+		return "", fmt.Errorf("chain genesis not initialized: %s", genesisPath)
+	}
+	genesisFile, err := os.ReadFile(genesisPath)
+	if err != nil {
+		return "", err
+	}
+	return string(genesisFile), nil
+}
+
+func formatChainInfo(ctx context.Context, c *networkchain.Chain, launchID uint64) (string, error) {
+	home, err := c.Home()
+	if err != nil {
+		return "", err
+	}
+	id, err := c.ID()
+	if err != nil {
+		return "", err
+	}
+
+	info := struct {
+		LaunchID  uint64 `json:"LaunchID"`
+		ChainID   string `json:"ChainID"`
+		Name      string `json:"Name"`
+		SourceURL string `json:"SourceURL"`
+		Hash      string `json:"Hash"`
+		HomePath  string `json:"HomePath"`
+	}{
+		LaunchID:  launchID,
+		ChainID:   id,
+		Name:      c.Name(),
+		SourceURL: c.SourceURL(),
+		Hash:      c.SourceHash(),
+		HomePath:  home,
+	}
+	return yaml.Marshal(ctx, info)
+}
+
+func formatChainAccounts(ctx context.Context, n network.Network, launchID uint64) (string, error) {
```

Comment:
This can handled in a separate PR but maybe we can also show the vesting accounts along with the normal genesis accounts